webapp/src/components/ClassOverlapTable.test.tsx

```diff
@@ -0,0 +1,77 @@
+import { fireEvent, screen, waitFor, within } from "@testing-library/react";
+import { renderWithRouterAndRedux } from "mocks/utils";
+import ClassOverlapTable from "./ClassOverlapTable";
+import { AvailableDatasetSplits } from "types/api";
+import { setupServer } from "msw/node";
+import { getClassOverlapAPIResponse } from "mocks/api/mockClassOverlapAPI";
+import { QueryPipelineState } from "types/models";
+
+const renderClassOverlapTable = (
+  availableDatasetSplits: AvailableDatasetSplits,
+  pipeline: QueryPipelineState
+) =>
+  renderWithRouterAndRedux(
+    <ClassOverlapTable
+      jobId="local"
+      pipeline={pipeline}
+      availableDatasetSplits={availableDatasetSplits}
+    />,
+    { route: "/local?pipeline_index=0", path: "/:jobId?:pipeline" }
+  );
+
+describe("ClassOverlapTable", () => {
+  const handlers = [getClassOverlapAPIResponse];
+  const server = setupServer(...handlers);
+
+  beforeAll(() => server.listen());
+  afterEach(() => server.resetHandlers());
+  afterAll(() => server.close());
+
+  it("should have expected columns", () => {
+    renderClassOverlapTable({ train: true, eval: true }, { pipelineIndex: 0 });
+    const expectedColumnHeaders = [
+      "Source Class",
+      "Target Class",
+      "Semantic Overlap Score",
+      "Pipeline Confusion",
+      "Utterances with Overlap",
+    ];
+    const columnHeaderDescription = [
+      "The class label of the samples being analyzed.",
+      "The class that the source class may look like. For pipeline confusion, this is the prediction.",
+      "Class overlap measures the extent to which source class samples are semantically similar to target class samples, in the training data. The score comes from both the proportion of samples and their degree of similarity.",
+      "Pipeline confusion indicates whether source class samples in the evaluation set are predicted to be in the target class.",
+      "Percent of source class samples that semantically overlap the target class (all in the training set).",
+    ];
+    const actualColumnHeaders = screen.getAllByRole("columnheader");
+    expectedColumnHeaders.forEach((name, index) => {
+      expect(actualColumnHeaders[index].textContent).toEqual(name);
```

Comment:
Detail: should we use this?
```suggestion
      expect(actualColumnHeaders[index]).toHaveTextContent(name);
```

Reply:
textContent does the exact match where [toHaveTextContent](https://github.com/testing-library/jest-dom#tohavetextcontent) needs regex to perform that. I am fine doing both. I just modified all the [places](https://github.com/ServiceNow/azimuth/pull/420/commits/676e106b55737fc6656283eb4096220e71b06dc3) that have "toHaveTextContent" to textContent.

Reply:
Ah-ha! So it's equivalent to `.toContain()` instead of `.toEqual()`. That's a very good point! I think you're right to favor exact match, then. You can ignore my suggestion, then.